src/BusinessLayer/CommunicationService.cpp

```diff
@@ -1,54 +1,42 @@
 #include <QIODevice>
-#include <QSerialPort>
 #include <QStringList>
 #include "CommunicationService.h"
 #include "InternetPeripheral.h"
 #include "SerialPortPeripheral.h"
 #include "SerialView.h"
+#include "InternetView.h"
 
-/*--------------------------------------------------------
-                Communication Service
-    Initially sets up with a SerialPort connection unless
-    specified otherwise.
---------------------------------------------------------*/
+#include <QDebug>
 
-CommunicationService::CommunicationService(SerialView& view)
-    : view_(view)
+CommunicationService::CommunicationService(SerialView& serialView, InternetView& internetView)
+    : serialView_(serialView)
+    , internetView_(internetView)
 {
-    setPeripheralSerialPort();
-    connect(&view_, SIGNAL(attemptConnectionSignal()), this, SLOT(attemptSerialConnection()));
-}
-
-void CommunicationService::setPeripheralSerialPort()
-{
-    outputDevice_ = new QSerialPort();
-    serialPeripheral_ = new SerialPortPeripheral(*outputDevice_);
-}
-
-void CommunicationService::setInternetConnection()
-{
-    //TODO: Implement or replace this function to set up a connection with the server
+    connect(&serialView_, SIGNAL(attemptConnectionSignal()), this, SLOT(attemptSerialConnection()));
+    connect(&internetView_, SIGNAL(attemptConnectionSignal()), this, SLOT(attemptInternetConnection()));
 }
 
 void CommunicationService::sendSerialData(const unsigned char* packet, int packetLength)
 {
-    serialPeripheral_->sendSerialData(packet, packetLength);
+    serialPeripheral_.sendSerialData(packet, packetLength);
 }
 
-void CommunicationService::sendInternetData(const QByteArray& data)
+void CommunicationService::sendInternetData(const QString& data)
 {
-    Q_UNUSED(data);
-    //TODO: Implement or replace this function to send incoming data
+    internetPeripheral_.sendInternetData(
+        internetView_.getExchangeName(),
+        internetView_.getRoutingKey(),
+        data);
 }
 
 void CommunicationService::attemptSerialConnection()
 {
-    QStringList paramList = (QStringList() << view_.getCommunicationPort());
-    serialPeripheral_->setParameters(paramList);
-    view_.setConnectionStatus(serialPeripheral_->attemptConnection());
+    serialView_.setConnectionStatus(serialPeripheral_.attemptConnection(serialView_.getCommunicationPort()));
 }
 
 void CommunicationService::attemptInternetConnection()
 {
-    //TODO: Implement or replace this function to try to connect to the server
+    internetView_.setConnectionStatus(internetPeripheral_.attemptConnection(
+                                          internetView_.getIpAddress(), internetView_.getPort()));
+    qDebug() << "CommunicationService::attemptInternetConnection";
```

Comment:
This debug statement could be removed. Or optionally I think it should be the first thing within the function if you still want to keep it (which is fine as there is not a whole lot of feedback when hitting connect).